tools/backup_analysis.py

```diff
@@ -0,0 +1,186 @@
+#!/usr/bin/env python3
+#
+# Public Domain 2014-present MongoDB, Inc.
+# Public Domain 2008-2014 WiredTiger, Inc.
+#
+# This is free and unencumbered software released into the public domain.
+#
+# Anyone is free to copy, modify, publish, use, compile, sell, or
+# distribute this software, either in source code form or as a compiled
+# binary, for any purpose, commercial or non-commercial, and by any
+# means.
+#
+# In jurisdictions that recognize copyright laws, the author or authors
+# of this software dedicate any and all copyright interest in the
+# software to the public domain. We make this dedication for the benefit
+# of the public at large and to the detriment of our heirs and
+# successors. We intend this dedication to be an overt act of
+# relinquishment in perpetuity of all present and future rights to this
+# software under copyright law.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+# EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+# MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+# IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+
+# Compare WT data files in two different home directories.
+
+import fnmatch, glob, os, sys, time
+
+def usage_exit():
+    print('Usage: backup_analysis.py dir1 dir2 [granularity]')
+    print('  dir1 and dir2 are POSIX pathnames to WiredTiger backup directories.')
+    print('  dir1 is assumed to be an older/earlier backup to dir2.')
+    print('Options:')
+    print('  granularity - an (optional) positive integer indicating the granularity')
+    print('  size in integer number of bytes of the incremental backup blocks.')
+    print('  The default is 16777216 (16MB).')
+    sys.exit(1)
+
+def die(reason):
+    print('backup_analysis.py: error: ' + reason, file=sys.stderr)
+    sys.exit(1)
+
+# Check that the directory given is a backup. Look for the WiredTiger.backup file.
+def check_backup(mydir):
+    if not os.path.isdir(mydir):
+        return False
+    backup_file = mydir + "/WiredTiger.backup"
+    if not os.path.exists(backup_file):
+        return False
+    return True
+
+def compare_file(dir1, dir2, filename, cmp_size, granularity):
+    f1_size = os.stat(os.path.join(dir1, filename)).st_size
+    f2_size = os.stat(os.path.join(dir2, filename)).st_size
+    min_size = min(f1_size, f2_size)
+    # Initialize all of our counters per file.
+    bytes_gran = 0
+    gran_blocks = 0
+    num_blocks = min_size // cmp_size
```

Comment:
The variable naming is a bit unfortunate, as there are two different kinds of blocks, 4KB and 16MB.

Would it make sense to always use `gran_blocks` vs. `cmp_blocks` in variable names, to make it clear? For example, `num_blocks` would become `num_cmp_blocks`. I think that would help. Maybe also you could rename `partial` to something like `cmp_block_partial` or `cmp_partial` to make it clearer.

Alternatively, you could refer to the 16MB blocks by some other name, such as "chunks."

Just an idea to improve clarity...

Reply:
I've made these changes. Thanks.